### PR TITLE
Minor bits and pieces

### PR DESCRIPTION
**Pull request by TrueBrain**
Just some bits and pieces to bring this repository on-par with the other Python repositories we have. Mostly moving around of code. This PR introduces no functional change what-so-ever.

### CODE REVIEW

File: webclient/pages/login.py [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
 import flask
 
+from ..main import (
+    api_get,
+    api_post,
+    app,
+    external_url_for,
+    redirect,
+)
+from ..main import (
```
Comment (frosch123) [inline webclient/pages/login.py]: Line 9 and 10 can be omitted.
Originally there were only two imports to fulfill the line length requirement.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+
+jobs:
+  flake8:
+    name: Flake8
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Flake8 bananas_api
```
Comment (frosch123) [inline .github/workflows/testing.yml:16]: bananas_frontend?

File: webclient/session.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+)
+
+auth_backend = {"method": "developer", "username": "frosch"}
+
+_sessions = dict()
+SESSION_COOKIE = "bananas_sid"
+MAX_SESSION_AGE = datetime.timedelta(hours=16)
+MAX_CSRF_AGE = datetime.timedelta(minutes=30)
+
+app = flask.Flask("webclient")
```
Comment (frosch123) [inline webclient/session.py:17]: import app, instead of creating twice

File: webclient/main.py [hunk 4]
```diff
@@ -1,200 +1,3 @@
...
-_api_url = "http://localhost:8080"  # TODO environment or something
-_frontend_url = "https://localhost:5000"  # TODO
-auth_backend = {"method": "developer", "username": "frosch"}
-
-_sessions = dict()
-SESSION_COOKIE = "bananas_sid"
-MAX_SESSION_AGE = datetime.timedelta(hours=16)
-MAX_CSRF_AGE = datetime.timedelta(minutes=30)
 
 app = flask.Flask("webclient")
```
Comment (frosch123) [inline webclient/main.py:3]: most flask examples even put this into __init__.py

**Comment by frosch123** [comment]
I am not sure how the configuration will look like, but now the config variables are split over two files.

File: webclient/main.py (hunk 4 above)
Comment (TrueBrain) [reply webclient/main.py:3]: I am not a fan of putting code in an `__init__.py`. People in general don't look there.

File: wishlist.txt [hunk 5]
```diff
@@ -1,16 +1,5 @@
...
-api:
-* meta info for new package goes to package? it should
-* set version members to "" should delete them (so that package data becomes active). None is used for "unchanged".
-* get download-url
-* license/changelog/readme-url
-* new list of licenses (cc 4.0), deprecated licenses (cc 3.0)
-* dependencies only editable for ai/gs, automatic for other content
-* endpoints to get list of licenses and branches
-* upload updates to savegames and heightmaps, keeping the unique-id
+- config parameters: api-url, frontend-url, mode: github, devauth, static
```
Comment (frosch123) [inline wishlist.txt:1]: already done with next commit

File: webclient/session.py [hunk 6]
```diff
@@ -41,7 +53,7 @@ def create_csrf_token(self, context):
             return None
 
         token = secrets.token_hex(8)
-        expires = datetime.datetime.utcnow() + MAX_SESSION_AGE
+        expires = datetime.datetime.utcnow() + max_session_age
```
Comment (frosch123) [inline webclient/session.py:56]: was wrong before, but max_csrf_age :)

Also, do you like leading _ for module variables?

**Pull request merged**